packages/block-library/src/query/edit/index.js

```diff
@@ -52,51 +43,4 @@ const QueryEdit = ( props ) => {
 	);
 };
 
-function PatternSelectionModal( {
-	clientId,
-	attributes,
-	setIsPatternSelectionModalOpen,
-} ) {
-	const { replaceBlock, selectBlock } = useDispatch( blockEditorStore );
-	const onBlockPatternSelect = ( blocks ) => {
-		const { newBlocks, queryClientIds } = getTransformedBlocksFromPattern(
-			blocks,
-			attributes
-		);
-		replaceBlock( clientId, newBlocks );
-		if ( queryClientIds[ 0 ] ) {
-			selectBlock( queryClientIds[ 0 ] );
-		}
-	};
-	// When we preview Query Loop blocks we should prefer the current
```

Comment:
We need to preserve this behavior. What this does in a few words is that shows the suggested patterns using the current `postType` set. For example if we have `postType: products` the insert/replace pattern flows show patterns with `products`.

Reply:
Thanks, good catch. I've added this back. Is there an easy way to test this? Maybe using WooCommerce?

Reply:
You can add the below in `query/variations.js` and the search for `Pages variation` and then click `choose`.

```
{
	name: 'pages-list',
	title: __( 'Pages variation' ),
	description: __(
		'Display a list of your most recent...'
	),
	icon: postList,
	attributes: {
		query: {
			perPage: 4,
			pages: 1,
			offset: 0,
			postType: 'page',
			order: 'desc',
			orderBy: 'date',
			author: '',
			search: '',
			sticky: 'exclude',
			inherit: false,
		},
	},
	scope: [ 'inserter' ],
},
```

Reply:
Thanks, this seems to be working correctly with the latest change:

<img width="1058" alt="Screenshot 2023-01-25 at 10 42 40 AM" src="https://user-images.githubusercontent.com/1464705/214654296-3b697f8c-0ece-44fb-bbaf-62987dcf05d6.png">